src/System.Text.RegularExpressions/src/System/Buffers/MemoryOrPinnedSpan.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    internal unsafe readonly struct MemoryOrPinnedSpan<T>
+    {
+        private readonly ReadOnlyMemory<T> _memory;
+        private readonly char* _pinnedSpan;
```

Comment:
This is a strange name for a `char*`.  It needn't be a pinned span; it could be any source of a ptr to fixed memory.